src/cargo/util/context/target.rs

```diff
@@ -53,6 +54,17 @@ pub(super) fn load_target_cfgs(
     let target: BTreeMap<String, TargetCfgConfig> = gctx.get("target")?;
     tracing::debug!("Got all targets {:#?}", target);
     for (key, cfg) in target {
+        if let Ok(Platform::Cfg(cfg_expr)) = key.parse() {
+            cfg_expr.walk_expr(|e| match e {
+                CfgExpr::True | CfgExpr::False => {
+                    if !gctx.cli_unstable().cfg_boolean_literals {
```

Comment:
nit: Should this condition be moved out of the new if-let?  If we know its not going to fail, why bother doing all of this?